### PR TITLE
Adds server-side apply docs to kpt live apply

### PR DESCRIPTION
**Pull request by seans3**
* Adds a section in the `kpt live appy` docs to describe server-side apply.
* Adds a section in the `kpt live preview` docs to describe server-side dry-run.
* Includes description of `--server-side`, `--field-manager`, and `--force-conflicts` flags, which can be used for both `apply` and `preview`.

### CODE REVIEW

File: site/content/en/reference/live/apply/_index.md [hunk 1]
```diff
@@ -22,12 +22,13 @@ it consistent with other kpt commands.
...
-|                     | `kubectl apply`            | `kpt live apply`          |
-|---------------------|----------------------------|---------------------------|
-|Usage                | kubectl apply -f /dir      | kpt live apply /dir       |
-|Applied resource set | Not tracked                | Tracked                   |
-|Prune                | Imperative and error prone | Declarative and reliable  |
-|Status               | Not supported              | Supported                 |
+|                     | `kubectl apply`                          | `kpt live apply`                        |
+|---------------------|------------------------------------------|-----------------------------------------|
+|Usage                | kubectl apply -f /dir                    | kpt live apply /dir                     |
+|                     | kubectl apply --server-side -f /dir      | kpt live apply --server-side /dir       |
```
Comment (mortent) [inline site/content/en/reference/live/apply/_index.md:28]: I don't think we need to cover this here. This is to highlight the pruning and status functionality.

File: site/content/en/reference/live/apply/_index.md [hunk 2]
```diff
@@ -36,6 +37,20 @@ group.  `kpt live apply` tracks the state of your applied resource set and
 related configuration. This helps `kpt` to reliably reconcile the real world
 resources with your configuration changes.
 
+### Client-Side Apply versus Server-Side Apply
+
+kpt live apply defaults to client-side apply, so the patches used to
+update resources are calculated on the client-side. In order to have
+the server calculate the patches, the same as `kubectl` add the
```
Comment (mortent) [inline site/content/en/reference/live/apply/_index.md:44]: Maybe rephrase the last part of this sentence.
I also don't think we need to cover SSA in detail here, we can add pointers to the kubernetes docs that describes it in detail. Mentioning the default field owner seems useful though.

**Review by mortent — COMMENTED** [review]
I think we also want to update the reference list of support flags for apply, preview... The flags for apply is at the bottom of the doc already in this PR.

File: site/content/en/reference/live/apply/_index.md [hunk 3]
```diff
@@ -36,6 +36,15 @@ group.  `kpt live apply` tracks the state of your applied resource set and
...
 resources with your configuration changes.
 
+### Client-Side Apply versus Server-Side Apply
+
+kpt live apply defaults to client-side apply, so the updates are accomplished
+by calculating and sending a patch from the client. Server-side apply
+with the `--server-side` flag sends the entire resource to the server
+for the update. The server-side flags and functionality are the same
+as kubectl. More information on
+[kubectl server-side apply]: <https://kubernetes.io/docs/reference/using-api/server-side-apply/>
```
Comment (mortent) [inline site/content/en/reference/live/apply/_index.md:46]: We usually put the links at the bottom of the page.

**Review by mortent — COMMENTED** [review]
I think this affect at least preview in addition to apply? We should update the description there too.

File: site/content/en/reference/live/apply/_index.md (hunk 3 above)
Comment (seans3) [reply site/content/en/reference/live/apply/_index.md]: Fixed.

Comment (seans3) [reply site/content/en/reference/live/apply/_index.md]: Added documentation in the `preview` command

File: site/content/en/reference/live/preview/_index.md [hunk 4]
```diff
@@ -47,5 +49,9 @@ DIRECTORY:
 ```
 --destroy:
   If true, dry-run deletion of all resources.
+
+--server-side:
```
Comment (mortent) [inline site/content/en/reference/live/preview/_index.md:53]: We should add `--field-manager` and `--force-conflicts` here too.

**Review by mortent — COMMENTED** [review]
Small comments, but otherwise looks good.

File: site/content/en/reference/live/preview/_index.md (hunk 4 above)
Comment (seans3) [reply site/content/en/reference/live/preview/_index.md:53]: Done.

File: site/content/en/reference/live/apply/_index.md (hunk 3 above)
Comment (seans3) [reply site/content/en/reference/live/apply/_index.md]: Resolved by adding documentation for these two additional flags.

**Review by mortent — APPROVED** [review]

File: site/content/en/reference/live/apply/_index.md [hunk 5]
```diff
@@ -269,8 +277,22 @@ DIR:
   This determines the output format of the command. The default value is
   events, which will print the events as they happen. The other option is
   table, which will show the output in a table format.
+
+--server-side:
```
Comment (frankfarzan) [inline site/content/en/reference/live/apply/_index.md:281]: Do we want to specify what versions of the api version support this? What's the error message otherwise?

Comment (mortent) [reply site/content/en/reference/live/apply/_index.md:281]: @seans3 Can you look at this so we can get this PR merged?

Comment (seans3) [reply site/content/en/reference/live/apply/_index.md:281]: Updated with version availability and error message. Please have a look

**Review by frankfarzan — APPROVED** [review]
